backend/src/main/java/uk/ac/cam/cl/kilo/Server.java

```diff
@@ -1,15 +1,62 @@
 package uk.ac.cam.cl.kilo;
 
-import static spark.Spark.*;
+import com.github.kevinsawicki.http.HttpRequest;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+import org.json.simple.parser.JSONParser;
 
+import java.io.*;
+import static spark.Spark.*;
 import java.security.SecureRandom;
 import java.util.Random;
+import uk.ac.cam.cl.kilo.data.Database;
+import uk.ac.cam.cl.kilo.data.User;
 
-public class Server {
+public class Server{
   private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray();
 
-  public static void main(String[] args) {
-    get("/hello", (req, res) -> "Hello World");
+  public static void main(String args[]){
+      String AppToken = "2332527273433130|12NQ8TuitMgVHpZ4pUpoZMjqqJ0";
+
+    get("/facebook/register", (request, response) -> {
+
+      String state = request.queryParamOrDefault("state","");
+      String code = request.queryParamOrDefault("code","");
+      String error = request.queryParamOrDefault("error","");
+
+
+       if(state == ""){ //stage1
+          state = generateState(32);
+         response.header("state",state);
+       }
+       else if(code ==""){//stage2
+         response.redirect("https://www.facebook.com/v3.2/dialog/oauth?&" +
+                 "client_id={2332527273433130}&" +
+                 "redirect_uri={"+request.host()+"/facebook/register}&"  +
+                 "state={"+state+"}");
+       }
+       else if(error != ""){//stage3
+         HttpRequest getAccessToken = HttpRequest.get("https://graph.facebook.com/v3.2/oauth/access_token", true,
+                 "client_id", "2332527273433130",
+                 "redirect_uri", request.host()+"/facebook/register",
+                 "client_secret", "13bb38aa6b63ffa248f0b3b15a6ba394",
```

Comment:
This needs to be revoked and when the new one is generated, this needs to go in a properties file